rewriter/AttrReader.cc

```diff
@@ -119,18 +119,61 @@ void ensureSafeSig(core::MutableContext ctx, const core::NameRef attrFun, ast::S
     }
 }
 
+ast::Send *findSendChecked(ast::Send *sharedSig) {
+    ENFORCE(ASTUtil::castSig(sharedSig), "We weren't given a send node that's a valid signature");
+
+    auto block = ast::cast_tree<ast::Block>(sharedSig->block);
+    auto body = ast::cast_tree<ast::Send>(block->body);
+
+    while (body != nullptr && body->fun != core::Names::checked()) {
+        body = ast::cast_tree<ast::Send>(body->recv);
+    }
+
+    return body;
+}
+
+// Heuristic to check if user-provided sig adds no runtime checking on top of an attr_reader.
+//
+// A user-provided sig causes an attr_reader method to behave differently from a normal attr_reader
+// method at runtime.
+//
+// If the answer would be "maybe adds runtime checking, but hard to tell," we answer that it does add
+// checking to be safe. Thus, the naming of this method is important: we can't rename this method to
+// `sigIsChecked` and negate all true/false. (Put another way: double negation elimination doesn't
+// apply here).
+bool sigIsUnchecked(core::MutableContext ctx, ast::Send *sig) {
```

Comment:
I like this comment, but perhaps we could have called it `sigCouldBeChecked`?  Then concerns about double negation go away.